File: gateway-service/src/main/java/com/movies/gateway/controller/MoviesController.java
package com.movies.gateway.controller;

import com.movies.common.movie.MovieTo;
import com.movies.gateway.movie.MoviesService;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@Controller
@AllArgsConstructor
public class MoviesController {
    private MoviesService moviesService;

    @GetMapping("/movie/{movieId}")
    public String movie(@PathVariable String movieId, Model model) {
        MovieTo movie = moviesService.getMovieById(movieId);
        model.addAttribute("movie", movie);
        return "movie_page";
    }
}

File: movies-list-service/src/main/kotlin/com/movies/list/controller/MoviesController.kt
package com.movies.list.controller

import com.movies.list.movies.Movie
import com.movies.list.movies.MoviesService
import com.movies.list.movies.to.CreateMoviesTo
import com.movies.list.utils.SecurityUtils.authUserId
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
class MoviesController(private val moviesService: MoviesService) {
    private val log = LoggerFactory.getLogger(MoviesController::class.java)

    @PostMapping("/admin/movies/add")
    fun addMovie(@RequestBody @Valid createMoviesTo: CreateMoviesTo): Movie {
        log.info("Admin ${authUserId()} add movie: $createMoviesTo")
        return moviesService.createMovie(createMoviesTo)
    }

    @GetMapping("/{movieId}")
    fun getMovieById(@PathVariable movieId: String): Movie {
        log.info("User get movie $movieId")
        return moviesService.getById(movieId)
    }
}